src/domains/queries.ml

```diff
@@ -101,8 +101,9 @@ type _ t =
   | IterVars: itervar -> Unit.t t
   | PathQuery: int * 'a t -> 'a t (** Query only one path under witness lifter. *)
   | DYojson: FlatYojson.t t (** Get local state Yojson of one path under [PathQuery]. *)
-  | HeapVar: VI.t t
+  | HeapVar: {on_stack: bool} -> VI.t t (* If on_stack is [true], then alloca() or a similar function was called *)
   | IsHeapVar: varinfo -> MayBool.t t (* TODO: is may or must? *)
+  | IsDynamicallyAlloced: varinfo -> MayBool.t t (* [true] if heap var represents dynamically alloced memory *)
```

Comment:
The naming is very confusing. If I understand correctly, `alloca` also uses `HeapVar`. But `alloca` doesn't allocate on the heap, it allocates on the stack!
And the difference between `IsHeapVar` and `IsDynamicallyAlloced` is completely unclear to me. Both `malloc` and `alloca` variables are created by `HeapVar`, so it seems like `IsHeapVar` is true for both. But both are also dynamically allocated.

Since these "heap" variables get `(alloc@...)` as their name, maybe their creation query should be `AllocVar` and the query corresponding to it `IsAllocVar`. Then `IsHeapVar` would distinguish those by `malloc` from those by `alloca`, or something.

Reply:
I agree that the naming might be confusing. `HeapVar` should be now referring to any dynamic allocation-site abstraction (i.e., `(alloca@...)`), which can in particular be `alloca` (on the stack) and `malloc` (on the heap).
Hence, I agree with the `AllocVar` name in place of `HeapVar`.

`IsDynamicallyAlloced` essentially checks if a var refers to a dynamic allocation-site abstraction, hence we could change it to `IsAllocVar`. `IsHeapVar` should be already be able to distinguish between `alloc@...` vars from `alloca` and from `malloc`. Iirc, I added an extra `vattr` to the `alloc@...` vars for this which is implemented in `HeapVar`.

In short, my suggestion:
* Rename `HeapVar` -> `AllocVar`
* Rename `IsDynamicallyAlloced` -> `IsAllocVar`
* Keep `IsHeapVar` as is
* (not sure if it'd be needed, but we could also add `IsStackVar` which is dual to `IsHeapVar`)

Please let me know, what do you think. CC @michael-schwarz, because we also discussed the naming with him before.

Reply:
Yes, seems reasonable!

Reply:
I did the renaming in https://github.com/goblint/analyzer/pull/1179/commits/7b1d29850036fb1287b10c8aeb962a162b7fac11